dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/SignalN/GeneratorPVTfoDiagramPQ_INIT.mo

```diff
@@ -0,0 +1,67 @@
+  import Dynawo;
+  import Dynawo.Electrical.Machines;
+  import Modelica.ComplexMath;
+
+  extends Machines.BaseClasses_INIT.BaseGeneratorParameters_INIT;
+  extends AdditionalIcons.Init;
+
+  parameter Types.ActivePower PMin "Minimum active power in MW (generator convention)";
+  parameter Types.ActivePower PMax "Maximum active power in MW (generator convention)";
+  parameter Types.ApparentPowerModule SNom "Nominal apparent power of the generator in MVA";
+  parameter Types.ReactivePower QNomAlt "Nominal reactive power of the generator in Mvar";
+  parameter Types.PerUnit RTfoPu "Resistance of the generator transformer in pu (base UNomHV, SNom)";
+  parameter Types.PerUnit XTfoPu "Reactance of the generator transformer in pu (base UNomHV, SNom)";
+  parameter Types.PerUnit rTfoPu "Ratio of the generator transformer in pu (base UBaseHV, UBaseLV)";
+  parameter Types.VoltageModulePu URef0Pu "Start value of the voltage regulation set point at terminal in pu (base UNom)";
+  parameter Types.ReactivePower QMin0 "Start value of minimum reactive power in Mvar (generator convention)";
+  parameter Types.ReactivePower QMax0 "Start value of maximum reactive power in Mvar (generator convention)";
+
+  Types.ActivePowerPu PMinPu "Minimum active power in pu (base SnRef) (generator convention)";
+  Types.ActivePowerPu PMaxPu "Maximum active power in pu (base SnRef) (generator convention)";
+  Types.ReactivePowerPu QMin0Pu "Start value of minimum reactive power in pu (base SnRef)";
+  Types.ReactivePowerPu QMax0Pu "Start value of maximum reactive power in pu (base SnRef)";
+  Types.VoltageModulePu UStatorRef0Pu "Start value of voltage regulation set point at stator in pu (base UNom)";
+  Types.VoltageModulePu UStator0Pu "Start value of voltage module at stator in pu (base UNom)";
+  Types.ComplexVoltagePu uStator0Pu "Start value of complex voltage at stator in pu (base UNom)";
+  Types.ComplexCurrentPu iStator0Pu "Start value of complex current at stator in pu (base UNom, SNom) (generator convention)";
+  Types.ComplexApparentPowerPu sStator0Pu "Start value of complex apparent power at stator in pu (base UNom, SNom) (generator convention)";
+  Types.ReactivePowerPu QStator0Pu "Start value of stator reactive power in pu (base QNomAlt) (generator convention)";
+
+protected
+  Types.ComplexVoltagePu uRef0Pu "Start value of complex voltage reference at terminal in pu (base UNom)";
+  Types.ComplexCurrentPu iRef0Pu "Start value of complex current reference at terminal in pu (base UNom, SnRef) (receptor convention)";
+
+equation
+  PMinPu = PMin / Dynawo.Electrical.SystemBase.SnRef;
+  QMin0Pu = QMin0 / Dynawo.Electrical.SystemBase.SnRef;
+  PMaxPu = PMax / Dynawo.Electrical.SystemBase.SnRef;
+  QMax0Pu = QMax0 / Dynawo.Electrical.SystemBase.SnRef;
```

Comment:
Should they be final parameters of the dynamic model instead?

Reply:
Could be, but I wanted the possibility to directly give the pu based parameters in the dynamic model which won't be possible with final parameters. This is what's done for the moment in all our DynaFlow generators models.

Reply:
Ok then. We can keep it the way it is